docs/cci_yml_reference.rst

```diff
@@ -0,0 +1,78 @@
+``cumulusci.yml`` Reference
```

Comment:
This doc isn't linked from anywhere, is it? Looks like it's not finished yet.

Reply:
Ya, l was thinking of keeping this but let's just go ahead and remove it since we aren't referencing it from anywhere.